source/adios2/common/ADIOSConfig.h.in

```diff
@@ -31,10 +31,12 @@
 
 /* Everything between here and the note above is programatically generated */
 
-#ifdef ADIOS2_HAVE_MPI
-#  define ADIOS2_USE_MPI 1
-#else
+#ifndef ADIOS2_USE_MPI
 #  define ADIOS2_USE_MPI 0
```

Comment:
When the application links to `adios2::{c,cxx11,fortran}_mpi` the default switches based on its usage requirements.  If we defaulted to `ADIOS2_HAVE_MPI` it would expose the MPI-based APIs in header files even though they are not linked.